libvmaf/tools/cli_parse.c

```diff
@@ -46,6 +47,7 @@ static const struct option long_opts[] = {
     { "cpumask",          1, NULL, ARG_CPUMASK },
     { "aom_ctc",          1, NULL, ARG_AOM_CTC },
     { "frame_cnt",        1, NULL, ARG_FRAME_CNT },
+    { "skip_ref_frame",   1, NULL, ARG_SKIP_REF_FRAME },
```

Comment:
It would be useful to slightly generalize the logic to allow negative values. For example, if the skip_ref_frame number is -2, it means to skip 2 dis frames instead. 

Once you have the logic ready, I can add two tests to cover both cases.

Reply:
It is a good question. I thought about this (actually to introduce another offset instead of being negative for distorted y4m). But if we assume that the distorted y4m is a decoded or reconstructed version of the reference, we cannot have frames in distorted y4m before the reference since we don't have reference to encode. Or do we have a use case to compare two arbitrary y4ms? If yes, then we need the negative offset.

Reply:
There were indeed use cases where you may need to skip some distorted frames (think of VMAF is computed on a shot-by-shot basis and for some reason the reference and distorted are a few frames misaligned). It would be useful to cover this case.

By the way, there is another way to achieve the frame skipping using [ffmpeg+libvmaf](https://github.com/Netflix/vmaf/blob/master/resource/doc/ffmpeg.md) combined with the select filter (example: `-vf select='gte(n\,0)*gte(6\,n)'`). I assume here you want to avoid calling ffmpeg.

Reply:
OK. I will add the support of skipping frames in a distorted sequence. I will use negative offset for this case, as you suggested unless you prefer a new parameter.

Regarding ffmpeg, yes, I'd like to avoid the call to ffmpeg. For our tests, we need a temporary y4m file for the conversion (cut a longer sequence to a shorter one and then calculate VMAF/PSNR) if source and distorted have different numbers of frames as the decoder we needed is not embedded in ffmpeg (like the case of AV2). If we are doing a lot of 4K tests at the same time, the peak disk space requirement would be quite huge.

Reply:
Maybe adding another parameter skip_dist_frame is better? If we use negative value to skip distorted frames, we can only have one skip offset, either for reference or for distorted, both not for both. If we add skip_dist_frame in addition to skip_ref_frame, we may have two different non-zero skip offsets at the same time. From the use case you mentioned, two parameters seem better. Your opinion? @li-zhi

Reply:
This is a good point. It's more general and less confusing. Let's have two parameters.

Reply:
--skip_dist_frame has been added. Please check.